### PR TITLE
Added docker support for frontend

### PR DESCRIPTION
**Pull request by llNovall**
I've added docker for frontend. It ran fine on my computer but let me know if there are any issues. I'm still learning docker. As for the docker-compose file in the frontend, the contents within it can later be moved to the similar named file in the root folder of this repository. Afterwards, just delete the compose file in the frontend folder.

### CODE REVIEW

File: Frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM node:20.7.0-alpine AS build
+WORKDIR /app
+COPY package*.json ./
+RUN npm install -g npm@10.2.5
+RUN npm ic
+COPY . .
+RUN npm run build
+FROM nginx:stable
+COPY --from=build /app/dist/frontend /usr/share/nginx/html
+EXPOSE 80
```
Comment (eddyizm) [inline Frontend/Dockerfile:10]: We are going to put this behind a reverse proxy so we don't want to expose port 80 directly, let's go with a higher port number.

File: Frontend/docker-compose.yml
Comment (eddyizm) [inline Frontend/docker-compose.yml:1]: This compose file needs to be merged and details added to the root compose file [here](https://github.com/eddyizm/NPI_FILES/blob/main/docker-compose.yml)

I am going to build a shell script to do this in podman as well.

File: Frontend/Dockerfile (hunk 1 above)
Comment (llNovall) [reply Frontend/Dockerfile:10]: How about 4200 then? Is that good?

File: Frontend/docker-compose.yml
Comment (llNovall) [reply Frontend/docker-compose.yml:1]: Sure, I'll merge the file.

File: Frontend/Dockerfile (hunk 1 above)
Comment (eddyizm) [reply Frontend/Dockerfile:10]: that works!

**Review by eddyizm — APPROVED** [review]

**Pull request merged**